FlowCal/plot.py

```diff
@@ -735,38 +1266,53 @@ def scatter3d(data_list,
     if not isinstance(color, list):
        color = [color]*len(data_list)
 
+    # Get transformation functions for each axis
+    # Explicit rescaling is required for non-linear scales because mplot3d does
+    # not natively support anything but linear scale.
+    if xscale == 'linear':
+        xscale_transform = lambda x: x
+    elif xscale == 'log':
+        xscale_transform = np.log10
+    elif xscale == 'logicle':
+        t = _LogicleTransform(data=data_list, channel=channels[0])
+        it = _InterpolatedInverseTransform(t, 0, t.M)
+        xscale_transform = it.transform_non_affine
+    else:
+        raise ValueError('scale {} not supported'.format(xscale))
+
+    if yscale == 'linear':
+        yscale_transform = lambda x: x
+    elif yscale == 'log':
+        yscale_transform = np.log10
+    elif yscale == 'logicle':
+        t = _LogicleTransform(data=data_list, channel=channels[1])
+        it = _InterpolatedInverseTransform(t, 0, t.M)
+        yscale_transform = it.transform_non_affine
+    else:
+        raise ValueError('scale {} not supported'.format(yscale))
+
+    if zscale == 'linear':
+        zscale_transform = lambda x: x
+    elif zscale == 'log':
+        zscale_transform = np.log10
+    elif zscale == 'logicle':
+        t = _LogicleTransform(data=data_list, channel=channels[2])
+        it = _InterpolatedInverseTransform(t, 0, t.M)
+        zscale_transform = it.transform_non_affine
+    else:
+        raise ValueError('scale {} not supported'.format(zscale))
+
     # Make 3d axis if necessary
     ax_3d = plt.gca(projection='3d')
 
     # Iterate through data_list
     for i, data in enumerate(data_list):
         # Get channels to plot
         data_plot = data[:, channels]
-        # Transform according to the scale of each axis
-        # Explicit rescaling is required for non-linear scales because mplot3d
-        # does not natively support anything but linear scale.
-        if xscale == 'linear':
-            data_plot_x = data_plot[:,0]
-        elif xscale == 'log':
-            data_plot_x = np.log10(data_plot[:,0])
-        else:
-            raise ValueError('scale {} not supported'.format(xscale))
-        if yscale == 'linear':
-            data_plot_y = data_plot[:,1]
-        elif yscale == 'log':
-            data_plot_y = np.log10(data_plot[:,1])
-        else:
-            raise ValueError('scale {} not supported'.format(yscale))
-        if zscale == 'linear':
-            data_plot_z = data_plot[:,2]
-        elif zscale == 'log':
-            data_plot_z = np.log10(data_plot[:,2])
-        else:
-            raise ValueError('scale {} not supported'.format(zscale))
         # Make scatter plot
-        ax_3d.scatter(data_plot_x,
-                      data_plot_y,
-                      data_plot_z,
+        ax_3d.scatter(xscale_transform(data_plot[:, 0]),
+                      yscale_transform(data_plot[:, 1]),
+                      zscale_transform(data_plot[:, 2]),
```

Comment:
Axes tick values are not shown by default, so this is not terribly important.

> this may be a lot of work for what is otherwise a very minor aesthetics issue.

Exactly. This can be a future issue, though.